src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.html

```diff
@@ -0,0 +1,67 @@
+<ng-container *ngIf="!loading; else isLoading">
+  <ng-container *ngIf="incompatible; else isCompatible">
+    <cd-alert-panel type="warning" i18n>The data received has the JSON format version 2.x and is currently incompatible with the dashboard.</cd-alert-panel>
+  </ng-container>
+  <ng-template #isCompatible>
+    <tabset *ngFor="let device of data | keyvalue">
+      <tab [heading]="device.value.device + ' (' + device.value.identifier + ')'">
+        <ng-container *ngIf="device.value.error; else noError">
+          <cd-alert-panel type="warning">{{ device.value.userMessage }}</cd-alert-panel>
+        </ng-container>
+        <ng-template #noError>
+          <div class="btn-group mb-3" role="group" aria-label="Report Actions">
+            <button class="btn btn-primary"
+                    title="Copy to Clipboard"
+                    i18n-title
+                    cdToClipboard
+                    [data]="device.value.report">
+              <i class="fa fa-files-o" aria-hidden="true"></i>
+            </button>
+            <button class="btn btn-primary"
+                    title="Download"
+                    i18n-title
+                    [cdTextToDownload]="device.value.report"
+                    [filename]="device.value.reportFilename">
```

Comment:
I'm curious: does this come from a real customer-driven requirement? I say it 'cause we are showing big amounts of tabular data (OSDs, Ceph settings, etc.) and I don't get why we do want this specific data to be downloadable (and not others too).

I just tried to check whether our ngx-datatables support proper copy-paste into tabular format (from dashboard into Google Spreadsheets) and that unfortunately doesn't work off the shelf.

I have 2 proposals around this:
- As technically the file to be downloaded should have the same contents as the REST API endpoint `get_smart_data`, why not enforcing the use of the exiting REST API for doing this?
- Why not extending copy+paste capabilities of all datatables/keyvalue tables.

Reply:
Interesting questions.

I've got some use cases in my mind (nothing that came from a customer). When a drive is either

1. about to fail and I want to verify that,
2. I'm unsure about it's status, or
3. smartmontools doesn't know the device,

I want to share a report, preferably in it's common format. That's the text representation of the report. I prefer to have the common format because:

1. I don't want to introduce our own format for smartctl data and
2. just the data of a single table is not so useful without information about the drive. For instance, `Raw read error rate` is [typically useless](https://www.smartmontools.org/wiki/FAQ#WhatdetailscanbeinterpretedfromRawreaderrorrate) when there's not documentation for it. The text representation contains the data from the `Device information` tab, the `SMART` tab and the overall health assessment panel.

Hence, I'd like to have the capabilities of this PR even if there was a download and copy-to-clipboard on our data tables.

As we concluded on our standup, it makes sense to implement a general download and copy-to-clipboard feature for all data tables. I created a [tracker issue](https://tracker.ceph.com/issues/41989) for that.

Reply:
Understood. My question was more about the use case behind this. If full copy+paste/downloads are required on a regular basis, probably they'll need to happen for a few/all OSD devices, and in that case relying on the REST API + some scripting would be the best way to obtain this data automatically.

Regarding information hiding: datatable provides column visibility toggles, so that could be a way to alleviate the data overloading (hiding by default the least relevant columns).

Thanks for the tracker issue!